### ISSUE TITLE
Cross-Origin Request Blocked When request made as text/html

### ISSUE THREAD

**Issue by mrkhanakia** [barryvdh/laravel-cors#123]
Hi
I have laravel route which returns HTML not JSON so when i return HTML it gives error
`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://local.pma/api/auth/signin. (Reason: CORS header 'Access-Control-Allow-Origin' missing).`

But if i return response as JSON then it works ok.

So i did a little research and tried to modify HandleCors.php and added that line on top
header("Access-Control-Allow-Origin: *");

![dc3n6il](https://cloud.githubusercontent.com/assets/2174170/16937611/2698767a-4d8e-11e6-9d67-7eea6d527bc3.png)

So basically it seems html content is not passing to $this->cors->addActualRequestHeaders function becuase if i put die() jsut before that function i still get HTML so $response = $next($request);
is directly returning HTML without passing it to above function.

Please resolve that issue

**Comment by barryvdh** [comment]
How are you returning your text/html? With a proper response object?

**Comment by mrkhanakia** [comment]
I am simply echoing <?php echo $html; ?>

**Comment by barryvdh** [comment]
In your view or in your controller? Don't ever echo stuff in your controllers.

**Comment by mrkhanakia** [comment]
but why echo will not work it is also returns html response. So how do i return HTML Content.
For example i have HTML Form which need to be return via Ajax Request and load it so how do i do that

**Comment by barryvdh** [comment]
`return $html`;

**Comment by mrkhanakia** [comment]
ok but how to debug 500 Error it shows nothing whenever i have 500 error
![tlylpki](https://cloud.githubusercontent.com/assets/2174170/16948013/f7ab22bc-4dcf-11e6-9040-0f06378cd98b.png)
